File: Go/codeSignal/firstDuplicate.go
package codesignal

func FirstDuplicate(a []int) int {
	dup := make(map[int]int)

	for i, v := range a {
		if _, ok := dup[v]; ok {
			return v
		}
		dup[v] = i
	}
	return -1
}

File: TypeScript/codeSignal/firstDuplicate.ts
function firstDuplicate(a: number[]): number {
  const dup = new Set()
  for (const v of a) {
    if (dup.has(v)) {
      return v
    }
    dup.add(v)
  }
  return -1
}



console.log(firstDuplicate([2, 1, 3, 5, 3, 2]))
